framework/app.mk

```diff
@@ -46,6 +46,13 @@ SRC_DIRS    := $(APPLICATION_DIR)/src
 PLUGIN_DIR  := $(APPLICATION_DIR)/plugins
 
 excluded_srcfiles += main.C
+excluded_srcfile_paths += $(shell find $(SRC_DIRS) -name main.C)
```

Comment:
Why is the name of this variable a "path"? It's a file. I think you intend for this to mean "relative_file_path". We have a list of excluded file names already (excluded_srcfiles). It's possible for an application to extend that list beyond just main.C. We put that in there for somebody I believe. You should probably be transforming all of the files in the excluded list to produce relative paths, not just main.C.

This file is getting a little out of control. There's yet a third place in this file where we deal with "main.C" again. Just trying to keep it from turning into complete spaghetti.

Reply:
> I think you intend for this to mean "relative_file_path".

So `excluded_srcfile_relative_paths`? Robert would appreciate the detail :-)

Reply:
Well I read that as paths and kept looking at it trying to figure out why you need to depend on a path. It threw me for a minute until I realized that it wasn't a path (translated as directory in my head). This app.mk file is getting unwieldy, not your fault, but we keep bolting onto it. Yeah Robert might call it "esrpath" or something.

Reply:
> (translated as directory in my head)

Yea you're right about that poor indication of meaning. I think I'll remove "paths" from the name